server.go

```diff
@@ -386,6 +386,46 @@ func (server *Server) createTLSConfig(entryPointName string, tlsOption *TLS, rou
 	// BuildNameToCertificate parses the CommonName and SubjectAlternateName fields
 	// in each certificate and populates the config.NameToCertificate map.
 	config.BuildNameToCertificate()
+	//Set the minimum TLS version if set in the config TOML
+	if server.globalConfiguration.EntryPoints[entryPointName].TLS.MinVersion != "" {
+		config.PreferServerCipherSuites = true
+		switch server.globalConfiguration.EntryPoints[entryPointName].TLS.MinVersion {
+		case "VersionTLS10":
+			config.MinVersion = tls.VersionTLS10
+		case "VersionTLS11":
+			config.MinVersion = tls.VersionTLS11
+		case "VersionTLS12":
+			config.MinVersion = tls.VersionTLS12
+		}
+	}
+	//Set the list of CipherSuites if set in the config TOML
+	if server.globalConfiguration.EntryPoints[entryPointName].TLS.CipherSuites != nil {
+		config.CipherSuites = make([]uint16, 0)
+		for _, cipher := range server.globalConfiguration.EntryPoints[entryPointName].TLS.CipherSuites {
+			switch cipher {
```

Comment:
Maybe this would be better to avoid this long switch case here for maintainability.
You could define a map with all the cipher suites:
```
var cipherSuites = map[string]uint16{
    `TLS_RSA_WITH_RC4_128_SHA`:     tls.TLS_RSA_WITH_RC4_128_SHA,
    `TLS_RSA_WITH_3DES_EDE_CBC_SHA`: tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA,
    ...
}
```
and then your code would be simply:
```
config.CipherSuites = []uint16{}
for _, cipher := range server.globalConfiguration.EntryPoints[entryPointName].TLS.CipherSuites {
 if exists, cipherConst := cipherSuites[cipher]; exists{
  config.CipherSuites = append(config.CipherSuites, cipherConst)
 } else {
  // log a warning
 }
}
```
WDYT?